src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -4742,9 +4794,75 @@ internal static object ThrowInvalidConversionException(object valueToConvert, Ty
                 return CacheConversion<object>(fromType, toType, LanguagePrimitives.ConvertIntegerToEnum, ConversionRank.Language);
             }
 
+            if (fromType == typeof(PSMethod) && toType.IsSubclassOf(typeof(MulticastDelegate)))
+            {
+                var psMethod = (PSMethod) fromInstance;
+
+                var maybeType = psMethod.instance as Type;
+                var methodInfoCandiates = maybeType != null
+                    ? maybeType.GetMethods(BindingFlags.Static|BindingFlags.Public)
+                    : psMethod.instance.GetType().GetMethods(BindingFlags.Instance | BindingFlags.Public);
+
+                var isAction = toType.FullName.StartsWith("System.Action`");
```

Comment:
What about delegates other than `Action` that return void?

Reply:
There's also just plain old non-generic `System.Action`.